src/Dfc.CourseDirectory.Core/DataStore/CosmosDb/QueryHandlers/DeleteApprenticeshipHandler.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using OneOf;
+using OneOf.Types;
+
+namespace Dfc.CourseDirectory.Core.DataStore.CosmosDb.QueryHandlers
+{
+    public class DeleteApprenticeshipHandler : ICosmosDbQueryHandler<Queries.DeleteApprenticeship, OneOf<NotFound, Success>>
+    {
+        public async Task<OneOf<NotFound, Success>> Execute(DocumentClient client, Configuration configuration, Queries.DeleteApprenticeship request)
+        {
+            var documentUri = UriFactory.CreateDocumentUri(
+                configuration.DatabaseId,
+                configuration.ApprenticeshipCollectionName,
+                request.ApprenticeshipId.ToString());
+
+            try
+            {
+                var result = await client.DeleteDocumentAsync(documentUri, new RequestOptions { PartitionKey = new PartitionKey(request.ProviderUkprn) });
```

Comment:
I know we're not supposed to do hard deletes, but https://github.com/SkillsFundingAgency/dfc-providerportal-apprenticeships/blob/main/src/Dfc.ProviderPortal.Apprenticeships/Helper/CosmosDbHelper.cs#L299